manifests/sharednet.pp

```diff
@@ -0,0 +1,22 @@
+# @summary
+#   Define a DHCP shared-network
+# @param subnets
+#   subnets to include in the shared-network statement
+# @param order
+#   Fragment order in the dhcpd.conf
+define dhcp::sharednet (
+  Hash[String,Hash] $subnets = {},
+  Integer[1] $order = 75,
+) {
+  concat::fragment { "dhcp.conf+${order}-start_${name}.dhcp":
+    target  => "${dhcp::dhcp_dir}/dhcpd.conf",
+    content => "shared-network ${name} {\n",
+    order   => "${order}-0start",
+  }
+  concat::fragment { "dhcp.conf+${order}-end_${name}.dhcp":
+    target  => "${dhcp::dhcp_dir}/dhcpd.conf",
+    content => "}\n",
+    order   => "${order}-9end",
```

Comment:
Probably I need to change 9end in Zend to ensure it is at the end os the of the dhcp::subnet resource